## 1. Importing libraries and datasets

In [22]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.linear_model import LinearRegression
from sklearn.feature_selection import RFE
from sklearn.model_selection import train_test_split

# Actualize SKLEARN
# from sklearn.impute import KNNImputer 

In [23]:
dataset = pd.read_csv(r'train.csv')
test_dataset = pd.read_csv(r'test.csv')

## 2. Separate Target from Dependent Variables

In [24]:
Y = dataset.iloc[:,-1]
X = dataset.iloc[:,:-1]

## 3. Pre processing and Select Variables

### 3.1. General knowledge of the datasets

In [25]:
X.dtypes

Patient_ID                             int64
Family_Case_ID                         int64
Severity                               int64
Name                                  object
Birthday_year                        float64
Parents or siblings infected           int64
Wife/Husband or children infected      int64
Medical_Expenses_Family                int64
Medical_Tent                          object
City                                  object
dtype: object

In [26]:
X.describe()
test_dataset.describe()

Patient_ID  Family_Case_ID    Severity  Birthday_year  \
count   400.000000      400.000000  400.000000     318.000000   
mean   1100.500000    46024.932500    2.247500    1989.707547   
std     115.614301    39279.856807    0.844409      14.171527   
min     901.000000     1162.000000    1.000000    1944.000000   
25%    1000.750000    13487.000000    1.000000    1980.250000   
50%    1100.500000    26721.500000    3.000000    1993.000000   
75%    1200.250000    79045.000000    3.000000    1999.000000   
max    1300.000000   128562.000000    3.000000    2019.000000   

       Parents or siblings infected  Wife/Husband or children infected  \
count                    400.000000                         400.000000   
mean                       0.400000                           0.452500   
std                        0.998746                           0.910944   
min                        0.000000                           0.000000   
25%                        0.000000                           0.000000   
50%                        0.000000                           0.000000   
75%                        0.000000                           1.000000   
max                        9.000000                           8.000000   

       Medical_Expenses_Family  
count               400.000000  
mean               1013.870000  
std                1587.588114  
min                   0.000000  
25%                 221.000000  
50%                 406.000000  
75%                 883.250000  
max               14345.000000

### 3.2. Identify and Fill/Remove Missing Values

In [29]:
def is_null(X):
    print("# of missing values by variable:")
    print(X.isna().sum())
    
is_null(X)
is_null(test_dataset)

# of missing values by variable:
Patient_ID                             0
Family_Case_ID                         0
Severity                               0
Name                                   0
Birthday_year                        177
Parents or siblings infected           0
Wife/Husband or children infected      0
Medical_Expenses_Family                0
City                                   2
dtype: int64


##### Medical Tent

We will drop the variable "Medical Tent", once it as 702 missing values from a total of 900 (78 %)

In [28]:
def drop_variables (X, variables):
    
    return X.drop(variables, axis = 1)
    
X = drop_variables(X, ['Medical_Tent'])
test_dataset = drop_variables(test_dataset, ['Medical_Tent'])

##### City

 To fill the missing values in the variable "City", we decide to use the mode, since there are only two observations missing city

In [65]:
X.City.value_counts()
X['City'] = X['City'].fillna(X['City'].mode()[0])
test_dataset['City'].fillna(X['City']).mode()[0])

##### Birthday Year

In what concerns the remaining missing values, all from the variable "Birthday Year", we decided to apply the K-Nearest-Neighbor algorithm to fill them. This decision was based in the fact that there are 177 missing values, which we consider too much for apllying a simple input (such as mean or median input), but not that many too remove a variable that we consider that might have some importance in our model. 
Later, with more knowledge of the dataset we might consider remove it.

In [32]:
# USE KNN for birthday Year input after restart anaconda w/ most recent sklearn version
# imputer = KNNImputer(n_neighbors=2)
# X_filled_knn = imputer.fit_transform(X)

### 3.3. Remove Outliers

In [31]:
X.City.value_counts()

Santa Fe       649
Albuquerque    169
Taos            80
Name: City, dtype: int64

### 3.5. Transform Variables

In [ ]:
X.loc[X['City'] == 'Santa Fe', 'Santa Fe'] = 1
X.loc[X['City'] != 'Santa Fe', 'Santa Fe'] = 0

X.loc[X['City'] == 'Albuquerque', 'Albuquerque'] = 1
X.loc[X['City'] != 'Albuquerque', 'Albuquerque'] = 0

X.loc[X['City'] == 'Taos', 'Taos'] = 1
X.loc[X['City'] != 'Taos', 'Taos'] = 0



X.drop(columns = ['City'], inplace = True)

### 3.4. Check Correlations

In [ ]:

corr_matrix=numerical.corr()
mask=np.zeros_like(corr_matrix, dtype=np.bool)
mask[np.triu_indices_from(mask)]=True
sb.heatmap(data=corr_matrix, mask=mask, center=0, annot=True, linewidths=2, cmap='coolwarm')
plt.tight_layout()

In [1]:
fagearogfn

NameError: name 'fagearogfn' is not defined